### PR TITLE
Add more details in the DeAI overview page

### PR DESCRIPTION
**Pull request by ulan**
This adds a "What is DeAI?" section that explains various flavors of term "DeAI"

### CODE REVIEW

**Comment by ulan** [comment]
@ielashi @jessiemongeon1 : ptal

File: docs/developer-docs/ai/overview.mdx [hunk 1]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 Decentralized AI, or DeAI for short, refers to the intersection between AI and blockchain technology.
 The term denotes a wide range of applications, starting from peripheral elements such as tokenization or decentralized marketplaces, and going all the way to running AI models fully on-chain as smart contracts.
 
+## What is DeAI?
+
+Similar to term "Web3", people may mean and refer to different things with term "DeAI".
+In some cases, the term is diluted and used for platforms and applications are neither decentralized nor trustworthy.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:17]:
```suggestion
The term "DeAI" may be used to refer to or describe a variety of different things. In some cases, the term is diluted and used by platforms and applications that are neither decentralized nor trustworthy. In other contexts, such as the context of ICP, DeAI refers to artificial intelligence running on a fully decentralized and secure platform. 
```

File: docs/developer-docs/ai/overview.mdx [hunk 2]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+
+Here is a list of various DeAI applications ordered from the strongest to weakest form of decentralization:
+
+1. **Fully on-chain training and inference**:
+This is the truest form of DeAI that brings the security and trustworthiness guarantees of smart contracts to DeAI applications.
+Supporting this use case is the long-term vision of ICP that will become possible after integrating GPU-enabled nodes.
+
+2. **Fully on-chain inference on models trained off-chain**:
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:25]:
```suggestion
2. **Fully on-chain inference using models trained off-chain**:
```

File: docs/developer-docs/ai/overview.mdx [hunk 3]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+This assumes that the model was trained off-chained and then uploaded on-chain.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:26]:
```suggestion
This assumes that the model was trained off-chain and then uploaded on-chain.
```

File: docs/developer-docs/ai/overview.mdx [hunk 4]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+Inference happens fully on-chain and has the same security and trustworthiness guarantees as regular smart contracts.
+ICP currently supports this use case for models with millions of parameters.
+
+3. **Storing the model on-chain and running inference on user's device**:
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:30]:
```suggestion
3. **Storing the model on-chain and running inference on a user's device**:
```

File: docs/developer-docs/ai/overview.mdx [hunk 5]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+This assumes that the model was trained off-chained and then uploaded on-chain.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:31]:
```suggestion
This assumes that the model was trained off-chain and then uploaded on-chain.
```

File: docs/developer-docs/ai/overview.mdx [hunk 6]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+Inference happens on user's device after downloading the model.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:32]:
```suggestion
Inference happens on the user's device after downloading the model.
```

File: docs/developer-docs/ai/overview.mdx [hunk 7]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+If the user trusts their own device, then they can trust that the inference ran correctly.
+A disadvantage here is that the model needs to be downloaded to user's device with corresponding drawbacks for confidentiality of the model and for user experience due to increased latency.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:34]:
```suggestion
A disadvantage here is that the model needs to be downloaded to the user's device with corresponding drawbacks of less confidentiality of the model and decreased user experience due to increased latency.
```

File: docs/developer-docs/ai/overview.mdx [hunk 8]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+ICP supports this use case for practically all existing models because a smart contract on ICP can store models up to 400GiB.
+
+4. **Tokenization, marketplaces, orchestration**:
+This is about using smart contracts as the tokenization, marketplace, and orchestration layer for AI models and AI hardware.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:38]:
```suggestion
This refers to using smart contracts as the tokenization, marketplace, and orchestration layer for AI models and AI hardware.
```

File: docs/developer-docs/ai/overview.mdx [hunk 9]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+Since ICP is a general-purpose blockchain, it supports such smart contracts of arbitrary complexity.
+
+5. **Smart contracts calling Web2 AI services**:
+Smart contract running on ICP can make HTTP requests to Web2 services including services like OpenAI and Claude.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:42]:
```suggestion
Smart contracts running on ICP can make HTTP requests to Web2 services including OpenAI and Claude.
```

File: docs/developer-docs/ai/overview.mdx [hunk 10]
```diff
@@ -11,16 +11,49 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
...
+See [an example](https://github.com/peterpeterparker/juno-openai) of a smart contract calling to OpenAI API.
```
Comment (jessiemongeon1) [inline docs/developer-docs/ai/overview.mdx:43]:
```suggestion
See [an example](https://github.com/peterpeterparker/juno-openai) of a smart contract calling the OpenAI API.
```

**Review by jessiemongeon1 — APPROVED** [review]

**Pull request merged**